### PR TITLE
Add ability to view all measures on one screen

### PR DESCRIPTION
**Pull request by sebbacon**
Per conversation with @Bengoldacre 

image

I'd like to think a bit more about the wording and arrangement; and switching to a custom single-measure view needs its own throbber, I think.  But would be good to deploy first for discussion.

I realise this is probably quite a hard PR to review and I've not gone back and interactively rebased a couple of odd WIP commits, but I *hope* it's clear enough for our purposes!

### CODE REVIEW

File: apps/base.py [hunk 1]
```diff
@@ -122,3 +124,13 @@ def toggle_entity_id_list_from_click_data(click_data, entity_ids):
...
+
+
+def filter_entity_ids_for_type(entity_type, entity_ids):
+    if entity_type == "ccg_id":
+        rx = r"^[0-9]{2}[A-Z]{1}$"
+    elif entity_type == "practice_id":
+        rx = r"^[A-Z]{1}[0-9]{5}$"
+    else:
+        rx = r"^[a-z]+$"
+    return [x for x in entity_ids if re.match(rx, x)]
```
Comment (evansd) [inline apps/base.py]: This worries me a bit. We've been here before and it hasn't ended well. Given that we already have a definitive list of the valid IDs for each type I think we could use that without baking in assumptions about the structure of the codes. I'm happy to do that if you like.

Also, random `re` module tip: you can create a regex using `rx = re.compile(<regex-string>)` and then match using `rx.match(x)` which can give slightly cleaner code than passing the string around (especially if you want to apply specific flags to it) and saves the re engine having to repeatedly recompile the expression.

File: apps/linecharts.py [hunk 2]
```diff
@@ -71,33 +69,22 @@ def get_decile_traces(df, col_name):
...
+    Returns an array of traces; a chart title; hint text giving user
+    pointers on what to do next; and an array annotations used for
+    explaining the legend.
 
+    """
     numerators = page_state.get("numerators", [])
     denominators = page_state.get("denominators", [])
     result_filter = page_state.get("result_filter", [])
-    groupby = page_state.get("groupby", None)
+    groupby = col_name = page_state.get("groupby", None)
```
Comment (evansd) [inline apps/linecharts.py]: We should just drop `col_name` and use `groupby` everywhere in this function: we never change the value of either so there's no need for two variables.

File: apps/measures.json [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    "denominators": ["K"],
+    "result_filter": "over_range",
+    "description": "Potassium is commonly monitored as a renal function test. Hyperkalaemia is a serious medical condition which, if severe, requires urgent hospitalisation. However, test potassium levels can be significantly affected by in vitro effects, such as storage time and temperature, and whether the sample has been centrifuged before transport. This can affect the result, and the corresponding treatment of the patient. Seasonal peaks in this measure may indicate high rates of false positives."
+  },
+  {
+    "numerators": ["K"],
+    "denominators": ["per1000"],
+    "result_filter": "all",
+    "description": "Potassium is commonly monitored as a renal function test. It may be useful for practices to know if they are testing at a higher or lower rate than average."
+  },
```
Comment (evansd) [inline apps/measures.json]: This is a dupe of the first measure

**Review by evansd — APPROVED** [review]
Nice! There are a few nice-to-haves below which we can ignore if you just need to get it out quickly. Or I can tackle myself if you need to get your head round FDAA.

Otherwise I think this is a good feature to have, and a nicely readable implementation.

**Comment by evansd** [comment]
@sebbacon Ready to go if you think my changes look OK

**Comment by sebbacon** [comment]
Great, thanks!

**Pull request merged**